packages/asset-swapper/contracts/src/ShellSampler.sol

```diff
@@ -84,26 +90,37 @@ contract ShellSampler
         view
         returns (uint256[] memory takerTokenAmounts)
     {
-        // Initialize array of maker token amounts.
-        uint256 numSamples = makerTokenAmounts.length;
-        takerTokenAmounts = new uint256[](numSamples);
+        return _sampleApproximateBuys(
+            ApproximateBuyQuoteOpts({
+                makerTokenData: abi.encode(makerToken, pool),
+                takerTokenData: abi.encode(takerToken, pool),
+                getSellQuoteCallback: _sampleSellForApproximateBuyFromShell
+            }),
+            makerTokenAmounts
+        );
+    }
 
-        for (uint256 i = 0; i < numSamples; i++) {
-            try
-                IShell(pool).viewTargetSwap
-                    {gas: DEFAULT_CALL_GAS}
-                    (takerToken, makerToken, makerTokenAmounts[i])
-                returns (uint256 amount)
-            {
-                takerTokenAmounts[i] = amount;
-                // Break early if there are 0 amounts
-                if (takerTokenAmounts[i] == 0) {
-                    break;
-                }
-            } catch (bytes memory) {
-                // Swallow failures, leaving all results as zero.
-                break;
-            }
+    function _sampleSellForApproximateBuyFromShell(
```

Comment:
That fixes the `buys` for Shell and the Shell forks. 
Right now since we have that 0 "slippage" on the `buys` for some routes, if we use the `Shell` buy function `viewTargetSwap`, it rounds up (tiny higher approximation than the settlement `targetSwap`), hence the 100% reverts on buys. 

```
IncompleteFillBuyQuoteError({
  buyToken: '0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48',
  boughtAmount: 1000667315,
  buyAmount: 1000667445
})